packages/kbn-esql-validation-autocomplete/src/autocomplete/autocomplete.ts

```diff
@@ -303,7 +309,24 @@ function getSourcesRetriever(resourceRetriever?: ESQLCallbacks) {
   return async () => {
     const list = (await helper()) || [];
     // hide indexes that start with .
-    return buildSourcesDefinitions(list.filter(({ hidden }) => !hidden).map(({ name }) => name));
+    return buildSourcesDefinitions(
+      list
+        .filter(({ hidden }) => !hidden)
+        .map(({ name, dataStreams, title }) => {
+          return { name, isIntegration: Boolean(dataStreams && dataStreams.length), title };
+        })
+    );
+  };
+}
+
+function getDataSourcesListRetriever(resourceRetriever?: ESQLCallbacks) {
+  const helper = getSourcesHelper(resourceRetriever);
+  return async () => {
+    const list = (await helper()) || [];
+    return list.map(({ name, dataStreams }) => ({
+      name,
+      dataStreams,
+    }));
   };
 }
```

Comment:
And I am sorry, I thought I had understood what you are proposing here but I don't. Can you please be more specific?

Reply:
Sorry this wasn't clear.

My suggestion is to move [this logic](https://github.com/elastic/kibana/pull/184716/files#diff-2a96c81cce17aff8cb8f0254f313903c2312b770ae08b3f4b938af0824d2e0fcR860-R864) into `getDataSourcesListRetriever` and rename it to `getDatastreamsForIntegration`.

My thinking was that this will make the purpose of [this branch](https://github.com/elastic/kibana/pull/184716/files#diff-2a96c81cce17aff8cb8f0254f313903c2312b770ae08b3f4b938af0824d2e0fcR858) more clear (check if there's a source already written in the query, if so, extract the integration name and get all its data streams).

Reply:
Ahh, got it now. Done https://github.com/elastic/kibana/pull/184716/commits/cb109e1dbe64608eee45d1878249c6c3bdf90a67